core/injector/application_injector.hpp

```diff
@@ -600,19 +600,19 @@ namespace kagome::injector {
         injector.template create<sptr<crypto::Bip39Provider>>();
     auto random_generator = injector.template create<sptr<crypto::CSPRNG>>();
     auto genesis_config =
-        injector.template create<sptr<application::GenesisConfig>>();
+        injector.template create<sptr<application::ChainSpec>>();
 
+    auto path = config.keystore_path(genesis_config->id());
+    auto key_file_storage_res = crypto::KeyFileStorage::createAt(path);
+    if (not key_file_storage_res) {
+      common::raise(key_file_storage_res.error());
+    }
```

Comment:
IMO it is better to create binding for `KeyFileStorage` with lambda with this code and keep CryptoStore binding to CryptoStoreImpl. There should be less code with such approach